sbt-scalajs-bundler/src/main/scala/scalajsbundler/Webpack.scala

```diff
@@ -100,32 +108,162 @@ object Webpack {
     * @param log Logger
     * @return The generated bundles
     */
-  def bundle(
-    generatedWebpackConfigFile: File,
-    customWebpackConfigFile: Option[File],
-    webpackResources: Seq[File],
-    entries: Seq[(String, File)],
-    targetDir: File,
-    log: Logger
+  def libraries(
+      generatedWebpackConfigFile: File,
+      customWebpackConfigFile: Option[File],
+      webpackResources: Seq[File],
+      entryPointFiles: Seq[(String, File)],
+      entries: Seq[(String, File)],
+      targetDir: File,
+      log: Logger
   ): Seq[File] = {
 
     val configFile = customWebpackConfigFile
       .map(Webpack.copyCustomWebpackConfigFiles(targetDir, webpackResources))
       .getOrElse(generatedWebpackConfigFile)
 
-    log.info("Bundling the application with its NPM dependencies")
+    log.info("Bundling the applications NPM dependencies")
     Webpack.run("--config", configFile.absolutePath)(targetDir, log)
 
-    val bundles =
-      entries.map { case (key, _) =>
-        // TODO Support custom webpack config file (the output may be overridden by users)
-        targetDir / Webpack.bundleName(key)
-      }
-    bundles
+    entries.map {
+      case (k, v) =>
+        targetDir / Webpack.libraryFileName(k)
+    }
   }
 
+  val loaderScript: String =
+    """
+      |var exports = {};
```

Comment:
IMO, for Node.js versus Browser, the real solution is to separate this sbt plugin into two plugins: an `NpmPlugin` and a `WebpackPlugin` (or whatever they're called), the latter depending on the former. If you're only fetching dependencies for Node.js, use `NpmPlugin`. If you want to bundle up for the browser, use `WebpackPlugin`.

But if you use `globalFallback`, you're not doing bundling at all anymore. You will prevent a bundler from doing dce across the Scala.js module and its dependencies (in the "full" mode only, for example). You'd be doing dependency extraction and resolution, but not any kind of bundling.

Maybe that's what's needed, but I'm not sure it's the right call. At least for prod/fullOpt I would still expect scalajs-bundler to bundle it all together in one script.